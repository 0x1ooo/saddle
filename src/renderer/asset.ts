import isDev from 'electron-is-dev';
import path from 'path';

export default function asset(filename: string): string {
  if (!filename) {
    return filename;
  }

  if (!path.isAbsolute(filename)) {
    throw new Error(`asset filename must be absolute to root path, for example:
  "/img/logo.svg"
    `);
  }
  return isDev ? filename : path.resolve(__dirname, '..', path.relative('/', filename));
}
