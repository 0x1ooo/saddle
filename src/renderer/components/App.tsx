import { M2R, R2M } from '@common/ipc-protocol';
import { ipcRenderer } from 'electron';
import React from 'react';
import { hot } from 'react-hot-loader';
import { asset } from 'renderer/utils/assets-helper';
import './app.scss';

const App = () => (
  <div className="App">
    <header className="App-header">
      <img src={asset('/img/logo.svg')} className="App-logo" alt="logo" />
      <p>
        Ready to go!{' '}
        <span role="img" aria-label="cheers!">
          🍻️
        </span>
      </p>
      <p>
        Edit <code>src/App.tsx</code> and save to reload!
      </p>
      <a
        className="App-link"
        href="https://reactjs.org"
        target="_blank"
        rel="noopener noreferrer"
      >
        Learn React
      </a>
      <button type="button" onClick={start}>
        启动/重启
      </button>
      <button type="button" onClick={stop}>
        停止
      </button>
    </header>
  </div>
);

function start() {
  ipcRenderer.send(R2M.START_TROJAN);
}
function stop() {
  ipcRenderer.send(R2M.STOP_TROJAN);
}

ipcRenderer.on(M2R.TROJAN_BUSY, () => console.log('[trojan] busy'));
ipcRenderer.on(M2R.TROJAN_STARTED, () => console.log('[trojan] running'));
ipcRenderer.on(M2R.TROJAN_STOPPED, () => console.log('[trojan] stopped'));
ipcRenderer.on(M2R.TROJAN_ERROR, (_, err) => console.error('[trojan]', err));

export default hot(module)(App) as () => JSX.Element;
