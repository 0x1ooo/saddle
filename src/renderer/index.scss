body {
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  height: 100vh;
  position: relative;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
    monospace;
}

.window-frame {
  padding: 0 !important;
  margin: 0 !important;
  width: 100vw;
  height: 100vh;
  -webkit-app-region: drag;
}
.window-interact {
  -webkit-app-region: no-drag;
}
