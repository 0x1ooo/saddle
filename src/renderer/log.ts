import { makeLoggerOptions } from '@common/log';
import { ensureDirSync } from '@common/utils/fs';
import isDev from 'electron-is-dev';
import { configure, getLogger, shutdown } from 'log4js';
import path from 'path';

const logDir = isDev ? '/' : path.join(__dirname, '../log');
const logFilename = 'renderer.log';

function initialize() {
  ensureDirSync(logDir);
  const appenders = ['stdout', 'console'];
  if (!isDev) {
    appenders.push('file');
  }
  const options = makeLoggerOptions(logDir, logFilename, {
    appenders: {
      console: {
        type: 'console',
        layout: {
          type: 'pattern',
          pattern: '%[[%p] %d{yyyy/MM/dd hh:mm:ss}%]\n%m',
        },
      },
    },
    categories: {
      ui: {
        appenders,
        level: isDev ? 'ALL' : 'INFO',
      },
    },
  });
  configure(options);
}
async function flush() {
  return new Promise<void>((resolve) => {
    shutdown((err) => {
      if (err) {
        console.error('renderer logger flush error:', err); // eslint-disable-line no-console
      }
      resolve();
    });
  });
}

const log = {
  initialize,
  flush,
  ui: () => getLogger('ui'),
};
export default log;
