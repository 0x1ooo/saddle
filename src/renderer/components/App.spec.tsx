import { render } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

test('app page matches previous snapshot', () => {
  const component = renderer.create(<App />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('app page have an nice title', () => {
  const { getByText } = render(<App />);
  const titleEl = getByText('Ready to go! 🍻️');
  expect(titleEl).toBeTruthy();
});
