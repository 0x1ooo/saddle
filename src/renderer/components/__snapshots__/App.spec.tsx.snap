// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app page matches previous snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="/img/logo.svg"
    />
    <p>
      Ready to go!
       
      <span
        aria-label="cheers!"
        role="img"
      >
        🍻️
      </span>
    </p>
    <p>
      Edit 
      <code>
        src/App.tsx
      </code>
       and save to reload!
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
    <button
      onClick={[Function]}
      type="button"
    >
      启动/重启
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      停止
    </button>
  </header>
</div>
`;
